src/vm/peimage.cpp

```diff
@@ -909,73 +909,149 @@ PTR_PEImageLayout PEImage::GetLayoutInternal(DWORD imageLayoutMask,DWORD flags)
     {
         _ASSERTE(HasID());
 
-        if (imageLayoutMask&PEImageLayout::LAYOUT_MAPPED)
+        BOOL bIsMappedLayoutSuitable;
+        BOOL bIsFlatLayoutSuitable;
+
+#ifdef PLATFORM_UNIX
+        bIsMappedLayoutSuitable = ((imageLayoutMask & PEImageLayout::LAYOUT_MAPPED) != 0);
```

Comment:
@gkhanna79, thank for your review.

I move the specified statements above, and the `!defined (PLATFORM_UNIX)` part now looks exactly as in your version.

If you don't mind, I would leave the "Suitable", instead of "Requested" in this case. This is because the `(MAPPED | FLAT)` "request" is not "create mapped AND flat layout"; it is "create mapped OR flat layout", so in this case both layouts are "suitable", but not both are requested - only one layout should be created.